src/np_codeocean/upload.py

```diff
@@ -83,6 +83,40 @@ def create_ephys_symlinks(session: np_session.Session, dest: Path,
         if not abs_path.is_dir():
             np_tools.symlink(as_posix(abs_path), dest / rel_path)
     logger.debug(f'Finished creating symlinks to raw ephys data files in {root_path}')
+    correct_structure(dest)
+
+def correct_structure(dest: Path) -> None:
+    """
+    In case some probes are missing, remove device entries from structure.oebin files with folders that don't actually exist.
+    """
+    logger.debug('Creating modified structure.oebin')
+    oebin_paths = dest.glob('**/*structure.oebin')
+    for oebin_path in oebin_paths:
+        logger.debug(f'Examining oebin: {oebin_path} for correction')
+        oebin_obj = np_tools.read_oebin(np_config.normalize_path(oebin_path.readlink()))
+
+        # iterate over copy of list so as to not disrupt iteration when elements are removed
+        for device in [device for device in oebin_obj['events']]:
+            events_dir = oebin_path.parents[0] / 'events'
```

Comment:
using the `.parent` attribute instead will allow to not have to index into an array but i don't view this as a very big deal.